python/lancedb/table.py

```diff
@@ -660,6 +660,56 @@ def compact_files(self, *args, **kwargs):
         For most cases, the default should be fine.
         """
 
+    @abstractmethod
+    def add_columns(self, transforms: Dict[str, str]):
+        """
+        Add new columns with defined values.
+
+        This is not yet available in LanceDB Cloud.
+
+        Parameters
+        ----------
+        transforms: Dict[str, str]
+            A map of column name to a SQL expression to use to calculate the
+            value of the new column. These expressions will be evaluated for
+            each row in the table, and can reference existing columns.
+        """
+
+    @abstractmethod
+    def alter_columns(self, alterations: Iterable[Dict[str, str]]):
```

Comment:
```suggestion
    def alter_columns(self, alterations: Iterable[Dict[str, Any]]):
```

That being said, why use a dictionary here instead of a `ColumnAlteration` data class?  That way it would be consistent with the `nodejs` lib

Reply:
In `nodejs`, the `ColumnAlteration` is just a typescript interface. So I think the invocations will look similar:

```python
table.alter_columns(
   { 'path': 'id', 'rename': 'new_id' },
   { 'path': 'price', 'nullable': true },
)
```
versus:
```js
await table.alterColumns([
  { path: 'id', rename: 'new_id' },
  { path: 'price', nullable: true }
])
```

Lei already expressed he didn't like passing in explicit options classes in Python. "Feels too Java-like", apparently.